src/main/java/data_access/StockDataAccessObject.java

```diff
@@ -6,103 +6,157 @@
 import okhttp3.Request;
 import okhttp3.Response;
 import org.json.JSONObject;
+import utility.exceptions.RateLimitExceededException;
 
-import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.io.InputStream;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Scanner;
+import java.util.*;
 
+/**
+ * A DataAccessObject that retrieves real time stock data
+ */
 public class StockDataAccessObject implements StockDataAccessInterface {
     private static final String BASE_URL = "https://finnhub.io/api/v1";
+    private static final String TICKERS_FILE = "/config/tickers.txt";
+
     private final OkHttpClient client;
     private final String apiKey;
+    private final List<String> tickers;
 
     public StockDataAccessObject() {
         this.client = new OkHttpClient();
+        this.tickers = new ArrayList<>();
 
-        // Load .env.local file and get the API token
+        // Load .env.local file and get API token
         Dotenv dotenv = Dotenv.configure().filename(".env.local").load();
         this.apiKey = dotenv.get("STOCK_API_KEY");
 
-    }
+        // Fetch ticker data from resource file and store in list
+        try (InputStream inputStream = getClass().getResourceAsStream(TICKERS_FILE)) {
```

Comment:
`getClass().getResourceAsStream()` reads from the relative path of the runtime class. When loading a configuration file with known absolute path, it may be better to use `ClassLoader.getResourceAsStream()` with the file's absolute path. You can confirm with ChatGPT as I'm not 100% certain about this point.

Reply:
We have set the resources folder, so it's okay to call the relative path.